Katello/katello
# Fixes #34257 - cv docker tags unpullable after resync with tag limiting

#### What are the changes introduced in this pull request?
Now, during content indexing, only the `RepositoryDockerMetaTags` related to the repository being re-indexed are deleted as necessary.

I had to edit one of the test fixtures because it had an invalid repository label. This was stopping me from editing the tags whitelist in the middle of the test.

#### Considerations taken when implementing this change?
The bug overall is pretty simple.  When reindexing, the `RepositoryDockerMetaTags` were not being grouped by their respective repositories before being deleted.
A test was added so this wouldn't happen again.

#### What are the testing steps for this pull request?
1) Sync a docker repo
2) Add that docker repo to a content view and publish
3) Ensure you can pull from the CV docker repo
4) Edit the repository to have a tag whitelist ("Limit Sync Tags")
5) Re-sync
6) Ensure that the blacklisted tags cannot be pulled from the library+default org cv repo
7) Try pulling the blacklisted tags from the CV docker repo and notice that you cannot.  This is the bug
8) Patch in this PR
9) Remove the whitelist entirely from the docker repo
10) Repeat steps 1-6
11) Try pulling the blacklisted tags again from the CV docker repo. Notice that you can. This means the bug is fixed
